styx-scheduler-service/src/main/java/com/spotify/styx/StyxScheduler.java

```diff
@@ -480,7 +485,7 @@ private TriggerListener trigger(
     };
   }
 
-  private static Consumer<Workflow> workflowChanged(
+  private Consumer<Workflow> workflowChanged(
```

Comment:
Most other methods in this class seem to be static and get passed everything needed. Did you consider passing `time` into this method instead of making it an instance method?